src/api_routes_v4.rs

```diff
@@ -0,0 +1,392 @@
+          ),
+      )
+      .route("/federated_instances", get().to(get_federated_instances))
+      // Post
+      .service(
+        // Handle POST to /post separately to add the post() rate limitter
+        resource("/post")
+          .guard(guard::Post())
+          .wrap(rate_limit.post())
+          .route(post().to(create_post)),
+      )
+      .service(
+        scope("/post")
+          .route("", get().to(get_post))
+          .route("", put().to(update_post))
+          .route("/delete", post().to(delete_post))
+          .route("/remove", post().to(remove_post))
+          .route("/mark_as_read", post().to(mark_post_as_read))
+          .route("/mark_many_as_read", post().to(mark_posts_as_read))
+          .route("/hide", post().to(hide_post))
+          .route("/lock", post().to(lock_post))
+          .route("/feature", post().to(feature_post))
+          .route("/list", get().to(list_posts))
+          .route("/like", post().to(like_post))
+          .route("/like/list", get().to(list_post_likes))
+          .route("/save", put().to(save_post))
+          .route("/report", post().to(create_post_report))
+          .route("/report/resolve", put().to(resolve_post_report))
+          .route("/report/list", get().to(list_post_reports))
+          .route("/site_metadata", get().to(get_link_metadata)),
+      )
+      // Comment
+      .service(
+        // Handle POST to /comment separately to add the comment() rate limitter
+        resource("/comment")
+          .guard(guard::Post())
+          .wrap(rate_limit.comment())
+          .route(post().to(create_comment)),
+      )
+      .service(
+        scope("/comment")
+          .route("", get().to(get_comment))
+          .route("", put().to(update_comment))
+          .route("/delete", post().to(delete_comment))
+          .route("/remove", post().to(remove_comment))
+          .route("/mark_as_read", post().to(mark_reply_as_read))
+          .route("/distinguish", post().to(distinguish_comment))
+          .route("/like", post().to(like_comment))
+          .route("/like/list", get().to(list_comment_likes))
+          .route("/save", put().to(save_comment))
+          .route("/list", get().to(list_comments))
+          .route("/report", post().to(create_comment_report))
+          .route("/report/resolve", put().to(resolve_comment_report))
+          .route("/report/list", get().to(list_comment_reports)),
+      )
+      // Private Message
+      .service(
+        scope("/private_message")
+          .route("/list", get().to(get_private_message))
+          .route("", post().to(create_private_message))
+          .route("", put().to(update_private_message))
+          .route("/delete", post().to(delete_private_message))
+          .route("/mark_as_read", post().to(mark_pm_as_read))
+          .route("/report", post().to(create_pm_report))
+          .route("/report/resolve", put().to(resolve_pm_report))
+          .route("/report/list", get().to(list_pm_reports)),
+      )
```

Comment:
Consider moving thinks like distinguish and lock to the update enpoints, and moving things like save and like to new "actions" update endpoints.

Reply:
You mean `/comment/update/distinguish`? Not sure that makes sense, because edits use `/comment` directly and there is no `/comment/update`. And `/comment/action/` sounds too generic, it could cover anything.

Reply:
The idea is to use the `/comment` update endpoint for all field updates including `distinguished`, and add new endpoints for actions tables such as `/comment/actions` or `/comment_actions` that treat each action as a field to update. I'm less sure about the second idea.